Sources/AppStoreConnectCLI/Services/Operations/CreateBetaGroupOperation.swift

```diff
@@ -10,53 +10,41 @@ struct CreateBetaGroupOperation: APIOperation {
-        self.getAppsOperation = GetAppsOperation(options: .init(bundleIds: [options.appBundleId]))
+        getAppsOperation = GetAppsOperation(options: .init(bundleIds: [options.appBundleId]))
+        createBetaGroupEndpoint = { appId in
+            .create(
+                betaGroupForAppWithId: appId,
+                name: options.groupName,
+                publicLinkEnabled: options.publicLinkEnabled,
+                publicLinkLimit: options.publicLinkLimit,
+                publicLinkLimitEnabled: options.publicLinkLimit != nil
+            )
+        }
     }
 
     private typealias AppAttributes = AppStoreConnect_Swift_SDK.App.Attributes
     private typealias BetaGroupAttributes = AppStoreConnect_Swift_SDK.BetaGroup.Attributes
 
     func execute(with dependencies: CreateBetaGroupDependencies) -> AnyPublisher<BetaGroup, Error> {
-        let options = self.options
-
         let app = getAppsOperation
             .execute(with: .init(apps: dependencies.apps))
             .compactMap(\.first)
 
-        let appAndGroupAttributes = app
-            .flatMap { app -> AnyPublisher<(AppAttributes?, BetaGroupAttributes?), Error> in
-                let endpoint = APIEndpoint.create(
-                    betaGroupForAppWithId: app.id,
-                    name: options.groupName,
-                    publicLinkEnabled: options.publicLinkEnabled,
-                    publicLinkLimit: options.publicLinkLimit,
-                    publicLinkLimitEnabled: options.publicLinkLimit != nil
-                )
-
-                let betaGroupResponse = dependencies.createBetaGroup(endpoint)
-
-                return betaGroupResponse
-                    .map({ (app.attributes, $0.data.attributes) })
-                    .eraseToAnyPublisher()
-            }
-
-        let betaGroup = appAndGroupAttributes.map { appAttributes, groupAttributes -> BetaGroup in
-            BetaGroup(
-                appBundleId: appAttributes?.bundleId,
-                appName: appAttributes?.name,
-                groupName: groupAttributes?.name,
-                isInternal: groupAttributes?.isInternalGroup,
-                publicLink: groupAttributes?.publicLink,
-                publicLinkEnabled: groupAttributes?.publicLinkEnabled,
-                publicLinkLimit: groupAttributes?.publicLinkLimit,
-                publicLinkLimitEnabled: groupAttributes?.publicLinkEnabled,
-                creationDate: groupAttributes?.createdDate
-            )
+        let betaGroup = app.flatMap { app -> AnyPublisher<BetaGroup, Error> in
+            var betaGroup = BetaGroup(appId: app.id)
+            betaGroup.update(with: app.attributes)
+
+            return dependencies
+                .createBetaGroup(self.createBetaGroupEndpoint(app.id))
+                .map { response -> BetaGroup in
+                    betaGroup.update(with: response.data.attributes)
+                    return betaGroup
+                }
+                .eraseToAnyPublisher()
```

Comment:
```suggestion
        let betaGroup = app.flatMap { app -> AnyPublisher<BetaGroup, Error> in
            var betaGroup = BetaGroup(appId: app.id)
            betaGroup.update(with: app.attributes)

            return dependencies
                .createBetaGroup(self.createBetaGroupEndpoint(app.id))
                .map { BetaGroup(attributes: $0.data.attributes, app: App(id: app.id, attributes: app.attributes)) }
                .eraseToAnyPublisher()
```

Reply:
Yep, putting a nested app is a good idea